pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -0,0 +1,343 @@
+								if strings.Contains(revised.Data[url.Name], ",,") {
+									res := revised.Data[url.Name]
+									revised.Data[url.Name] = strings.ReplaceAll(res, ",,", ",")
+								}
+							}
+						}
+						newInsURLs := strings.Split(url.Value, ",")
+						for _, newEntry := range newInsURLs {
+							if !(strings.Contains(revised.Data[url.Name], newEntry)) {
+								if revised.Data[url.Name] == "" {
+									revised.Data[url.Name] = newEntry
+								} else {
+									str := []string{revised.Data[url.Name], newEntry}
+									revised.Data[url.Name] = strings.Join(str, ",")
+								}
+							}
+						}
+					} else {
+						//now add the incoming URLs as fresh entry
+						newInsURL := strings.Split(url.Value, ",")
+						for _, newEntry := range newInsURL {
+							if !(strings.Contains(revised.Data[url.Name], newEntry)) {
+								str := []string{revised.Data[url.Name], newEntry}
+								revised.Data[url.Name] = strings.Join(str, ",")
+							}
+						}
+					}
+				}
+			} else { //removing custom resource. Update configMap
+				//check if only dummy entery exist then no deletion can happen
+				if strings.Contains(revised.Data[url.Name], url.Value) {
+					revised.Data[url.Name] = strings.ReplaceAll(revised.Data[url.Name], url.Value, "")
+					if strings.HasPrefix(revised.Data[url.Name], ",") {
+						str := revised.Data[url.Name]
+						revised.Data[url.Name] = strings.TrimPrefix(str, ",")
+					}
+					if strings.HasSuffix(revised.Data[url.Name], ",") {
+						str := revised.Data[url.Name]
+						revised.Data[url.Name] = strings.TrimSuffix(str, ",")
+					}
+					if strings.Contains(revised.Data[url.Name], ",,") {
+						str := revised.Data[url.Name]
+						revised.Data[url.Name] = strings.ReplaceAll(str, ",,", ",")
+					}
+					if revised.Data[url.Name] == "" {
+						delete(revised.Data, url.Name)
+					}
+				}
+			}
+		}
+		if !reflect.DeepEqual(revised.Data, current.Data) {
+			current.Data = revised.Data
+			err = splctrl.UpdateResource(client, &current)
+			if err != nil {
+				return nil, err
+			}
+		}
+		return &current, nil
+	}
+
+	for _, url := range newURLs {
+		current.Data[url.Name] = url.Value
+	}
+
+	// Set name and namespace
+	current.ObjectMeta = metav1.ObjectMeta{
+		Name:      GetSplunkMonitoringconsoleConfigMapName(namespace, SplunkMonitoringConsole),
+		Namespace: namespace,
+	}
+
+	err = splctrl.CreateResource(client, &current)
+	if err != nil {
+		return nil, err
+	}
+
+	return &current, nil
+}
+
+func getConfigMapHash(client splcommon.ControllerClient, namespace string) (string, error) {
+
+	namespacedName := types.NamespacedName{Namespace: namespace, Name: GetSplunkMonitoringconsoleConfigMapName(namespace, SplunkMonitoringConsole)}
+	var current corev1.ConfigMap
+
+	err := client.Get(context.TODO(), namespacedName, &current)
+	// if configMap doens't exist and we try to get configMap hash just return with empty string
+	if err != nil {
+		return "", err
+	}
+
+	contents := current.Data //map[string]string
```

Comment:
Also, if data is nil or an empty map we could just return here.

Reply:
This is just an extra check, even if data is null or empty we create a hash of empty configMap